node/out_request.js

```diff
@@ -35,9 +36,14 @@ function TChannelOutRequest(id, options) {
     self.responseEvent = self.defineEvent('response');
     self.finishEvent = self.defineEvent('finish');
 
+    assert(options.channel, 'channel required');
+
     self.logger = options.logger;
     self.random = options.random;
     self.timers = options.timers;
+    self.retryCount = options.retryCount;
```

Comment:
we create one out request for every attempt. So this is either 0, 1, 2 etc. Every retry has a new OutRequest object.